packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -1,6 +1,47 @@
 // TODO: Put public facing types in this file.
 
-/// Checks if you are awesome. Spoiler: you are.
-class Awesome {
-  bool get isAwesome => true;
+import 'dart:io';
+
+import 'icln_plugin_base.dart';
+
+class Plugin implements ICLNPlugin {
+  // Data members
+  /*
+  ManifestMethod _manifest;
+  List<Interceptor> _preInterceptor;
+  List<Interceptor> _postInterceptor;
```

Comment:
pre and post interceptor is somethings that we can avoid, for now, what we should implement is the raw API described in the core lightning docs https://lightning.readthedocs.io/PLUGINS.html

The Java API have some wired stuff inside, that we can avoid to repet here!